depends/Makefile

```diff
@@ -293,7 +293,7 @@ $(host_prefix)/toolchain.cmake : toolchain.cmake.in $(host_prefix)/.stamp_$(fina
             -e 's|@no_sqlite@|$(NO_SQLITE)|' \
             -e 's|@no_upnp@|$(NO_UPNP)|' \
             -e 's|@no_natpmp@|$(NO_NATPMP)|' \
-            -e 's|@no_usdt@|$(NO_USDT)|' \
+            -e 's|@no_usdt@|$(NO_USDT)|' -e 's|@usdt_packages@|$(usdt_packages_)|' \
```

Comment:
Why do we need to do this

Reply:
Considering https://github.com/hebasto/bitcoin/blob/19722e3e728933e629eb8e407075fb193b2908e2/depends/packages/packages.mk#L28 the `usdt_packages` variable is non-empty for Linux only.

For other systems, in the absence of `NO_USDT=1`, the depends build system has `set(WITH_USDT ON)` in the toolchain file, which is wrong.

Reply:
If this is only relevant for Linux, then we should really fix the depends output based on package availability, rather than working around issues that shouldn't exist, but I'm also not sure if that's true when we get macOS support, in which case, I assume it's just a bug for Windows?

Reply:
> ... I'm also not sure if that's true when we get macOS support, in which case, I assume it's just a bug for Windows?

We can return back to this problem in the time when we actually get macOS support.

Reply:
MacOS (and FreeBSD) tracepoints don't use systemtap and don't require any other dependencies.

Reply:
Ok, we should fix this here then.

Reply:
> Ok, we should fix this here then.

I don't understand what exactly is broken. What are steps to reproduce the bug?

Reply:
> the depends build system has set(WITH_USDT ON) in the toolchain file, which is wrong.

Reply:
That is the behaviour without this line diff.

This PR actual behavior is correct. The depends build system has `set(WITH_USDT ON)` in the toolchain file only when the package is available.

Reply:
Yes my point is to fix depends. Rather than add a workaround for a bug that shouldnt exist.

Reply:
I apologise but I feel completely lost. I still don't understand what the bug is.

Is it present in the master branch?
Is it introduced in this PR?
What options do trigger this bug when invoking `make -C depends`?

Reply:
I just mean the fact that depends is handling anything USDT related at all, for non-linux systems. If this wasn't the case, we could have cleaner logic here. We can just track this as something to cleanup at some point, but the improvements to master would be relevant pre or post CMake (ideally before as that'll keep the build systems closer between branches).

Reply:
It looks to me like the current (master) behavior for a non-linux platform is:
- USDT is enabled by default
- the package doesn't get built by depends
- usdt is enabled by the config.site
- configure fails to find the headers so it's disabled in the end

On this branch that approach won't work, because it's opt-in only. Therefore @hebasto's solution is to only enable it if it's not disabled _and_ if it's defined for the platform. I think that's a reasonable generic approach.

Reply:
Though I think we could simply with:

`'s|@usdt_packages@|$(usdt_packages_$(NO_USDT))|'`

Then just test for:
```CMake
if("@usdt_packages@")
  set(WITH_USDT ON CACHE BOOL "")
```
?

And to @fanquake's point, I think that would apply to master as well:
```diff
diff --git a/depends/config.site.in b/depends/config.site.in
index 7c699087bbc..b6b38bd3148 100644
--- a/depends/config.site.in
+++ b/depends/config.site.in
@@ -70,7 +70,7 @@ if test -z "$enable_zmq" && test -n "@no_zmq@"; then
   enable_zmq=no
 fi

-if test -z "$enable_usdt" && test -n "@no_usdt@"; then
+if test -z "$enable_usdt" && test -n "@usdt_packages@"; then
   enable_usdt=no
 fi
```

Reply:
Thanks! [Reworked](https://github.com/hebasto/bitcoin/pull/162#issuecomment-2086714275).